masonry/src/properties/types/length.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use masonry_core::debug_panic;
+
+/// A value representing a width, height, or similar distance value.
+///
+/// Its value is always finite and non-negative.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Length {
+    value: f64,
+}
+
+impl Length {
+    /// A length of zero.
+    pub const ZERO: Self = Self { value: 0. };
+
+    /// Create a length, in logical pixels.
+    ///
+    /// # Panics
+    ///
+    /// If debug assertions are on, this will panic in these cases:
+    ///
+    /// - `value` is NaN.
+    /// - `value` is infinite.
+    /// - `value` is negative.
+    ///
+    /// If debug assertions are off, this will return zero instead of panicking.
+    pub fn px(value: f64) -> Self {
```

Comment:
Should this `track_caller`?

Reply:
I believe that it works on the individual implementations, rather than the trait method itself? I forget the details

Reply:
The reference actually sounds like it also works with traits: https://doc.rust-lang.org/reference/attributes/codegen.html#r-attributes.codegen.track_caller

> When applied to functions and methods in trait declarations, the attribute applies to all implementations.